packages/ember-glimmer/lib/environment.js

```diff
@@ -147,11 +151,17 @@ export default class Environment extends GlimmerEnvironment {
 
   getComponentDefinition(path, symbolTable) {
     let name = path[0];
+    let finalizer = _instrumentStart('render.compile', instrumentationPayload, name);
     let blockMeta = symbolTable.getMeta();
     let owner = blockMeta.owner;
     let source = blockMeta.moduleName && `template:${blockMeta.moduleName}`;
-
-    return this._definitionCache.get({ name, source, owner });
+    let definition = this._definitionCache.get({ name, source, owner });
+    if (definition) {
+      definition.finalizer = finalizer;
```

Comment:
@mixonic notes that this may not work as expected if the same component is used multiple times.